### PR TITLE
Retrieve Submissions and Student data from Google Classroom and display on the front end

### PR DESCRIPTION
**Pull request by rsampdev**

### CODE REVIEW

File: project/src/main/java/com/google/autograder/servlets/CreateNewAssignmentServlet.java [hunk 1]
```diff
@@ -34,78 +34,64 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
...
-        postBody.append("}");
-
-        return postBody.toString();
+        String year = request.getParameter("year");
+        String month = request.getParameter("month");
+        String day = request.getParameter("day");
+        String hours = request.getParameter("hours");
+        String minutes = request.getParameter("minutes");
+
+        JsonObject assignmentObject = new JsonObject();
```
Comment (mahimashanware) [inline project/src/main/java/com/google/autograder/servlets/CreateNewAssignmentServlet.java]: we have an assignment class currently, can you update that class and make the assignment object with that class? for consistency

File: project/src/main/java/com/google/autograder/servlets/ListAssignmentSubmissionsServlet.java [hunk 2]
```diff
@@ -36,36 +46,131 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
...
+        List<String> studentNames = new ArrayList<>();
+        List<String> studentEmails = new ArrayList<>();
+        List<String> studentUserIDs = new ArrayList<>();
+        List<String> studentSubmissionDriveFileLinks = new ArrayList<>();
+
+        parseStudentSubmissionInfo(json, studentNames, studentUserIDs, studentSubmissionDriveFileLinks);
+        parseStudentInfo(authorization, courseID, studentUserIDs, studentNames, studentEmails);
 
-        System.out.println("\n\n" + json + "\n\n");
+        JSONArray jsonArray = new JSONArray();
```
Comment (mahimashanware) [inline project/src/main/java/com/google/autograder/servlets/ListAssignmentSubmissionsServlet.java]: also add a class object for this

File: project/src/main/java/com/google/autograder/servlets/ListAssignmentSubmissionsServlet.java [hunk 3]
```diff
@@ -36,36 +46,131 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
...
+            JSONObject studentSubmissionData = new JSONObject();
+
+            studentSubmissionData.put("name", studentName);
+            studentSubmissionData.put("email", studentEmail);
+            studentSubmissionData.put("link", studentSubmissionDriveFileLink);
+
+            jsonArray.add(studentSubmissionData);
+        }
+
+        // storeAssignmentSubmissionsData(json, courseID, assignmentID);
```
Comment (mahimashanware) [inline project/src/main/java/com/google/autograder/servlets/ListAssignmentSubmissionsServlet.java]: probably can remove these comments if they aren't needed

**Review by mahimashanware — COMMENTED** [review]
For listAssignments and createAssignments, are we inputting the assignment info into datastore? It seems that the information that is pulled is directly from GC and shown on the front ebd but that causes issues later on. Is it possible to add it to datastore first and get that info from datastore instead?

File: project/src/main/java/com/google/autograder/servlets/CreateNewAssignmentServlet.java (hunk 1 above)
Comment (rsampdev) [reply project/src/main/java/com/google/autograder/servlets/CreateNewAssignmentServlet.java]: Right here I'm building out the post body for the create new assignment endpoint. All this data is sent to the Google Classroom API and not to the Database. When this step is completed we redirect to the assignments page where another servlet pulls out the data we need on our end and then adds it to the Database.

File: project/src/main/java/com/google/autograder/servlets/ListAssignmentSubmissionsServlet.java (hunk 2 above)
Comment (rsampdev) [reply project/src/main/java/com/google/autograder/servlets/ListAssignmentSubmissionsServlet.java]: +1

File: project/src/main/java/com/google/autograder/servlets/ListAssignmentSubmissionsServlet.java (hunk 3 above)
Comment (rsampdev) [reply project/src/main/java/com/google/autograder/servlets/ListAssignmentSubmissionsServlet.java]: +1

File: project/src/main/java/com/google/autograder/servlets/CreateNewAssignmentServlet.java (hunk 1 above)
Comment (mahimashanware) [reply project/src/main/java/com/google/autograder/servlets/CreateNewAssignmentServlet.java]: ahhh got it, makes sense

**Comment by rsampdev** [comment]
> For listAssignments and createAssignments, are we inputting the assignment info into datastore? It seems that the information that is pulled is directly from GC and shown on the front ebd but that causes issues later on. Is it possible to add it to datastore first and get that info from datastore instead?

For creating new assignments that data is sent to the Google Classroom API and we use another servlet to go between Google Classroom and the Database for assignments. The ListAssignmentsServlet handles that.

For listing assignment submissions, currently that data is being sent from Google Classroom straight to the front end. Right now I'm implementing the steps to save that data to the Database and then pull from the Database before displaying to the front end.

**Review by BenLin00 — DISMISSED** [review]

File: project/src/main/java/com/google/autograder/servlets/GetStudentServlet.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        try {
+            HttpURLConnection connection = (HttpURLConnection) new URL(endpoint).openConnection();
+                
+            connection.setRequestMethod("GET");
+            connection.setRequestProperty("Accept", "application/json");
+            connection.setRequestProperty("Authorization", authorization);
+                
+            json = API.getJSON(connection);
+        } catch(Exception e) {
+            return;
```
Comment (jonathanjlin) [inline project/src/main/java/com/google/autograder/servlets/GetStudentServlet.java:43]: You should also log the exception (e.printStackTrace()) so you know when something went wrong. In general, when you catch an exception, you should do something about it, typically logging or wrapping it in a different exception

**Review by jonathanjlin — DISMISSED** [review]

File: project/src/main/java/com/google/autograder/servlets/GetStudentServlet.java (hunk 4 above)
Comment (rsampdev) [reply project/src/main/java/com/google/autograder/servlets/GetStudentServlet.java:43]: +1

**Review by BenLin00 — APPROVED** [review]

**Pull request merged**